### PR TITLE
fix(amazonq): add field for builderId vs IdC

### PR DESCRIPTION
**Pull request by dhasani23**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Need to track if users using builderId or IdC for `/transform`. Had some redundant telemetry/logging. Deleted dead code too.

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt [hunk 1]
```diff
@@ -143,19 +143,13 @@ class CodeModernizerSession(
                 LOG.warn { "Job was cancelled by user before start job was called" }
                 CodeModernizerStartJobResult.Cancelled
             } else {
-                val errorMessage = "Failed to start job"
-                LOG.error(e) { errorMessage }
```
Comment (dhasani23) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt:147]: This was already being logged and emitted elsewhere

File: plugins/toolkit/jetbrains-rider/testData/NuGet.config
[the remark this answers is not in the document]
Comment (dhasani23) [reply plugins/toolkit/jetbrains-rider/testData/NuGet.config:1]: Removed

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt [hunk 2]
```diff
@@ -273,6 +267,8 @@ class CodeModernizerSession(
             ) { shouldStop.get() }
         } catch (e: Exception) {
             val errorMessage = "Unexpected error when uploading artifact to S3: ${e.localizedMessage}"
+            LOG.error { errorMessage }
```
Comment (mk-fan) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt:270]: Caller `createModernizationJob` seems to also log this error message. Will this be redundant?

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformTelemetryManager.kt [hunk 3]
```diff
@@ -6,8 +6,13 @@ package software.aws.toolkits.jetbrains.services.codemodernizer
 import com.intellij.openapi.components.Service
 import com.intellij.openapi.components.service
 import com.intellij.openapi.project.Project
+import io.ktor.network.sockets.connect
```
Comment (mk-fan) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformTelemetryManager.kt:9]: What is this import?

Comment (dhasani23) [reply plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformTelemetryManager.kt:9]: Whoops don't know how that got in there, removing now

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt (hunk 2 above)
Comment (dhasani23) [reply plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt:270]: See above, I removed it from there and put it here instead.

**Review by manodnyab — APPROVED** [review]

File: plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformTelemetryManager.kt [hunk 4]
```diff
@@ -183,10 +179,16 @@ class CodeTransformTelemetryManager(private val project: Project) {
...
 
     fun dependenciesCopied() = CodetransformTelemetry.dependenciesCopied(codeTransformSessionId = sessionId)
-    fun configurationFileSelectedChanged() = CodetransformTelemetry.configurationFileSelectedChanged(codeTransformSessionId = sessionId)
-    fun jobIsStartedFromUserPopupClick() = CodetransformTelemetry.jobIsStartedFromUserPopupClick(codeTransformSessionId = sessionId)
-    fun jobIsCanceledFromUserPopupClick() = CodetransformTelemetry.jobIsCanceledFromUserPopupClick(codeTransformSessionId = sessionId)
-    fun jobIsStartedFromChatPrompt() = CodetransformTelemetry.jobIsStartedFromChatPrompt(codeTransformSessionId = sessionId)
+    fun jobIsStartedFromChatPrompt() {
+        val connection = checkBearerConnectionValidity(project, BearerTokenFeatureSet.Q)
+        var authType: CredentialSourceId? = null
+        if (connection.connectionType == ActiveConnectionType.IAM_IDC && connection is ActiveConnection.ValidBearer) {
```
Comment (mk-fan) [inline plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformTelemetryManager.kt:185]: nit: Can probably check `connection is ActiveConnection.ValidBearer` as an outer if.

**Review by mk-fan — APPROVED** [review]

**Pull request merged**